code/game/objects/structures/fence.dm

```diff
@@ -73,6 +80,19 @@
 		if(human.species.can_shred(human))
 			attack_generic(human, 25)
 
+	if(!door || operating || cut)
+		return
+	operating = TRUE
+	playsound(loc, 'sound/effects/fenceopen.ogg', 25, 1)
+	update_icon()
+	sleep(opening_time)
```

Comment:
We don't use `sleep`, use `addtimer(CALLBACK(src, PROC_REF(name_of_continuation_proc), opening_time)`